### PR TITLE
[Storage]: Add a new command group `az storage share-rm` to use the Microsoft.Storage resource provider for Azure file share management operations.

### PR DESCRIPTION
**Pull request by jsntcy**
This PR is to support #11583 which added a new command group 'az storage share-rm". There are following commands in that command group:

- az storage share-rm create
- az storage share-rm delete
- az storage share-rm exists
- az storage share-rm show
- az storage share-rm list
- az storage share-rm update

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -620,6 +620,16 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('storage share') as c:
         c.argument('share_name', share_name_type, options_list=('--name', '-n'))
 
+    with self.argument_context('storage share-rm', resource_type=ResourceType.MGMT_STORAGE) as c:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: can we mark this command group in preview?

**Comment by yonzhan** [comment]
@jsntcy move to S164.

**Comment by Juliehzl** [comment]
let's find a time to discuss some implementation details offline according to will's feedback

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+**Storage**
+
+* Add a new command group `az storage share-rm` to use the Microsoft.Storage resource provider for Azure file share management operations.
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst]: What about "Add a new command group `az storage share-rm` to support manage File share (create/delete/update/show/list) in mgmt plane"?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 3]
```diff
@@ -625,6 +629,21 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('storage share') as c:
         c.argument('share_name', share_name_type, options_list=('--name', '-n'))
 
+    for item in ['create', 'delete', 'exists', 'list', 'show', 'update']:
+        with self.argument_context('storage share-rm {}'.format(item), resource_type=ResourceType.MGMT_STORAGE) as c:
+            c.argument('resource_group_name', required=False)
+            c.argument('account_name', storage_account_type)
+            c.argument('share_name', share_name_type, options_list=('--name', '-n'), id_part='child_name_2')
+            c.argument('share_quota', type=int)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: I will prefer use --quota to keep consistent with existing `az share create` command

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 4]
```diff
@@ -625,6 +629,21 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+            c.argument('metadata', nargs='+',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: According to the command guideline shared, * will be better.

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 5]
```diff
@@ -34,6 +34,10 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     t_queue_service = self.get_sdk('queue#QueueService')
     t_table_service = get_table_data_type(self.cli_ctx, 'table', 'TableService')
 
+    storage_account_type = CLIArgumentType(options_list='--storage-account',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: actually I will prefer keeping using --account-name. I get a typo when I want to use new commands due to convention.

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 6]
```diff
@@ -625,6 +629,21 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('storage share') as c:
         c.argument('share_name', share_name_type, options_list=('--name', '-n'))
 
+    for item in ['create', 'delete', 'exists', 'list', 'show', 'update']:
+        with self.argument_context('storage share-rm {}'.format(item), resource_type=ResourceType.MGMT_STORAGE) as c:
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: For all these commands, we should have api constraint, right?

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 7]
```diff
@@ -1176,6 +1176,79 @@
...
+short-summary: Update the properties for an Azure file share.
+examples:
+  - name: Update the properties for an Azure file share 'MyFileShare' under the storage account 'MyStorageAccount' (account name) in resource group 'MyResourceGroup'.
+    text: az storage share-rm update -g MyResourceGroup --storage-account MyStorageAccount --name MyFileShare --share-quota 3 --metadata key1=value1 key2=value2
+  - name: Update the properties for an Azure file share 'MyFileShare' under the storage account 'MyStorageAccount' (account id).
+    text: az storage share-rm update --storage-account MyStorageAccount --name MyFileShare --share-quota 3 --metadata key1=value1 key2=value2
+  - name: Update the properties for an Azure file shares by resource id.
+    text: az storage share-rm update --ids file-share-id --share-quota 3 --metadata key1=value1 key2=value2
+"""
+
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py]: storage account name must be lower case. can you change it for examples?

File: src/azure-cli/azure/cli/command_modules/storage/commands.py [hunk 8]
```diff
@@ -320,6 +326,18 @@ def get_custom_sdk(custom_module, client_factory, resource_type=ResourceType.DAT
         client_factory=file_data_service_factory,
         resource_type=ResourceType.DATA_STORAGE)
 
+    with self.command_group('storage share-rm', command_type=file_shares_mgmt_sdk,
+                            custom_command_type=get_custom_sdk('file',
+                                                               cf_mgmt_file_shares,
+                                                               resource_type=ResourceType.MGMT_STORAGE),
+                            resource_type=ResourceType.MGMT_STORAGE, min_api='2019-04-01', is_preview=True) as g:
+        g.command('create', 'create')
+        g.command('delete', 'delete')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/commands.py]: can we add confirmation for delete command?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 3 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: The help is from SDK: The maximum size of the share, in gigabytes. Must be greater than
                                   0, and less than or equal to 5TB (5120). For Large File Shares,
                                   the maximum size is 102400.

---
In reply to: [367792148](https://github.com/Azure/azure-cli/pull/11649#discussion_r367792148) [](ancestors = 367792148)

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 4 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: should be "+", it means we must provide one argument.

---
In reply to: [367792392](https://github.com/Azure/azure-cli/pull/11649#discussion_r367792392) [](ancestors = 367792392)

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 5 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: we'll use --storage-account and see if the users complain or not to determine if we need also use --account-name.

---
In reply to: [367793774](https://github.com/Azure/azure-cli/pull/11649#discussion_r367793774) [](ancestors = 367793774)

File: src/azure-cli/azure/cli/command_modules/storage/_help.py (hunk 7 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/storage/_help.py]: OK, will update it

---
In reply to: [367800069](https://github.com/Azure/azure-cli/pull/11649#discussion_r367800069) [](ancestors = 367800069)

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (jsntcy) [reply src/azure-cli/HISTORY.rst]: CLI users should not care if it's mgmt plane or data plane

---
In reply to: [364247847](https://github.com/Azure/azure-cli/pull/11649#discussion_r364247847) [](ancestors = 364247847)

File: src/azure-cli/azure/cli/command_modules/storage/commands.py (hunk 8 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/storage/commands.py]: OK, I'll update it

---
In reply to: [368257995](https://github.com/Azure/azure-cli/pull/11649#discussion_r368257995) [](ancestors = 368257995)

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 9]
```diff
@@ -1178,6 +1178,83 @@
...
+  - name: Create a new Azure file share 'MyFileShare' with metadata and quota as 10 GB under the storage account 'mystorageaccount' (account id).
+    text: az storage share-rm create --storage-account mystorageaccount --name MyFileShare --quota 10 --metadata key1=value1 key2=value2
+"""
+
+helps['storage share-rm delete'] = """
+type: command
+short-summary: Delete the specified Azure file share.
+examples:
+  - name: Delete an Azure file share 'MyFileShare' under the storage account 'mystorageaccount' (account name) in resource group 'MyResourceGroup'.
+    text: az storage share-rm delete -g MyResourceGroup --storage-account mystorageaccount --name MyFileShare
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py]: Share name should be in lowercase

**Pull request merged**